airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -140,6 +123,41 @@ class S3ConsumerFactory {
         }
     }
 
+    fun createAsync(
+        outputRecordCollector: Consumer<AirbyteMessage>,
+        storageOps: S3StorageOperations,
+        s3Config: S3DestinationConfig,
+        catalog: ConfiguredAirbyteCatalog
+    ): SerializedAirbyteMessageConsumer? {
+        val writeConfigs = createWriteConfigs(storageOps, s3Config, catalog)
+        val createFunction =
+            getCreateFunction(
+                s3Config,
+                Function<String, BufferStorage> { fileExtension: String ->
+                    FileBuffer(fileExtension)
+                },
+                DEFAULT_NAMESPACE
```

Comment:
why do we have a default_namespace constant here, while we have a parameter in flushBufferFunction? This method looks like the config `default_namespace` parameter is ignored, if it exists